packages/astro/src/@types/astro.ts

```diff
@@ -2166,6 +2166,35 @@ export interface AstroUserConfig {
 			 * ```
 			 */
 			schema?: EnvSchema;
+
+			/**
+			 * @docs
+			 * @name experimental.env.validateSecretsOnStart
+			 * @kind h4
+			 * @type {boolean}
+			 * @default `false`
+			 * @version 4.11.3
+			 * @description
+			 *
+			 * By default, public variables are validated on start (dev/build modes) and secrets at runtime only. If enabled, secrets will also be checked on start. This is useful for example in some CIs to make sure all your secrets are correctly set before deploying.
```

Comment:
```suggestion
			 * Whether or not to validate secrets on the server when starting the dev server or running a build.
			 *
			 * By default, only public variables are validated on the server at `buildStart`. Private variables are validated on the client at runtime only. If enabled, secrets will also be checked on start. This is useful in some continuous integration (CI) pipelines to make sure all your secrets are correctly set before deploying.
```

OK, I said a lot of things that I'm not sure are totally accurate ( :sweat_smile:  ) but if they are, I think this is a helpful amount of information to give! Can you make this correct?

Reply:
1. Ok, so what I want to clarify with "on dev and build" is whether it's correct to say, "when starting the dev server or running a build"

I know with the dev server running, it's *like* continuously updating, so I'm not sure whether it's correct to say "when STARTING the dev server" or "when RUNNING the dev server". Can you confirm EXACTLY, being overly pedantic, when/what is happening so I can consider my options here?

2. I chose "private variables" to contrast with "public variables" because we'd already used the word "secret" and I figured this was "another way for people to understand." Not opposed to changing it, but using both interchangeably can help add variety to the text people have to read... IF they mean the same thing!

Can you confirm whether there's a different meaning or nuance in saying "secret variables" vs "private variables", or something that you feel "secret" conveys that "private" doesn't? If I know how private is sub-optimal, then maybe there's another phrase I can use as an alternative description when I want some word variety to avoid repetition.

Reply:
1. I think it's technically "when STARTING the dev server"
2. I was not clear enough my bad! I don't really mind secret or private, my point was that "on the client" is wrong

Reply:
D'oh, gotcha!

Reply:
Ok, @florian-lefebvre - updated my suggestion, see whether you think that works!